.github/workflows/build.yaml

```diff
@@ -4,27 +4,51 @@ on:
   push:
     branches: [ rolling ]
   workflow_dispatch:
+  schedule:
+    # Run every morning to detect flakiness and broken dependencies
+    - cron: '03 5 * * *'
 defaults:
   run:
     shell: bash
 jobs:
-  test:
+  build_and_test_source_rolling:
```

Comment:
Instead of having three separate jobs like this, can we retain the `distro` matrix from earlier and setup one job only to run for each distro in the matrix? Within this job, we can have a special `if` condition to check if `distro == rolling` and if so, build from source. If you need access to both the ROS distro version and ubuntu version, the matrix can be defined [like so](https://github.com/open-rmf/rmf_ci_templates/blob/1ae7b4f64644ab2f8724e9a1e1b3d14803caa61e/.github/workflows/reusable_build.yaml#L13-L25).

Also it appears that your changes now pull a `rostooling/setup-ros-docker:ubuntu-x` image. How often is this image updated? And why not use `ros-tooling/setup-ros@v0.7` to setup from latest `ubuntu` base image [like so](https://github.com/ros-tooling/setup-ros?tab=readme-ov-file#setting-up-the-worker-installing-system-dependencies-and-ros-linux).

Reply:
Here is an example of using an `if` condition within the action: https://github.com/open-rmf/rmf_ci_templates/blob/1ae7b4f64644ab2f8724e9a1e1b3d14803caa61e/.github/workflows/reusable_build.yaml#L53